helm/nessie/templates/deployment.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "nessie.fullname" . }}
+  labels:
+    {{- include "nessie.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "nessie.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "nessie.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "nessie.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: NESSIE_VERSION_STORE_TYPE
+              value: {{ .Values.versionStoreType }}
+            {{- if eq .Values.versionStoreType "DYNAMO" }}
+            {{- if .Values.dynamodb.region }}
+            - name: AWS_REGION
+              value: {{ .Values.dynamodb.region }}
+            {{- end }}
+            - name: AWS_ACCESS_KEY_ID
```

Comment:
will these environment variables show up in logs anywhere? Or in a `docker ps` type command? I believe the image is set up to read from all the standard locations for AWS config so we could also write to `~/.aws/credentials` if these env variables would be visible anywhere.

Reply:
I'm not aware that those env vars would be visible. They will only show up on the Deployment/Pod like this:
```
    Environment:
      NESSIE_VERSION_STORE_TYPE:  DYNAMO
      AWS_REGION:                 us-west-2
      AWS_ACCESS_KEY_ID:          <set to the key 'aws_access_key_id' in secret 'awscreds'>      Optional: false
      AWS_SECRET_ACCESS_KEY:      <set to the key 'aws_secret_access_key' in secret 'awscreds'>  Optional: false
```

Reply:
actually inside the POD the secret will be exposed as a normal env var as described in https://kubernetes.io/docs/concepts/configuration/secret/#consuming-secret-values-from-environment-variables